python/HeaderService/hutils.py

```diff
@@ -50,30 +50,49 @@
 LOGGER = logging.getLogger(__name__)
 
 
-def create_logger(logfile=None, level=logging.NOTSET):
+def configure_logger(logger, logfile=None, level=logging.NOTSET, log_format=None, log_format_date=None):
 
     """
-    Simple logger creation used across modules using
-    the same definitions
+    Configure an existing logger
     """
-
-    LOGGER = logging.getLogger(__name__)
     # Define formats
-    FORMAT = '[%(asctime)s][%(levelname)s][%(name)s][%(funcName)s] %(message)s'
-    FORMAT_DATE = '%Y-%m-%d %H:%M:%S'
+    if log_format:
+        FORMAT = log_format
+    else:
+        FORMAT = '[%(asctime)s.%(msecs)03d][%(levelname)s][%(name)s][%(funcName)s] %(message)s'
+    if log_format_date:
+        FORMAT_DATE = log_format_date
+    else:
+        FORMAT_DATE = '%Y-%m-%d %H:%M:%S'
     formatter = logging.Formatter(FORMAT, FORMAT_DATE)
+
     handlers = []
     # Set the logfile handle if required
     if logfile:
         fh = RotatingFileHandler(logfile, maxBytes=2000000, backupCount=10)
         fh.setFormatter(formatter)
+        fh.setLevel(level)
         handlers.append(fh)
+        logger.addHandler(fh)
 
     # Set the screen handle
     sh = logging.StreamHandler(sys.stdout)
     sh.setFormatter(formatter)
+    sh.setLevel(level)
     handlers.append(sh)
-    logging.basicConfig(handlers=handlers, level=level, format=FORMAT, datefmt=FORMAT_DATE)
+    logger.addHandler(sh)
+
+    return handlers
```

Comment:
Is there any need to return the handlers? One can get them from the logger and it would simplify the code to not bother. I see you are using it below in create_logger but I suspect that is not needed. Certainly I have found that by adding a StreamHandler and setting its level I get the output I expect without calling basicConfig.

Reply:
Thanks for noticing it. I've fixed it.